src/Diagnostics.ModelsAndUtils/Models/ResponseExtensions/DetectorStatus.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Diagnostics.ModelsAndUtils.Models.ResponseExtensions
+{
+    public static class ResponseDetectorStatusExtensions
+    {
+        /// <summary>
+        /// Sets the status of a detector
+        /// </summary>
+        /// <param name="response">Response object</param>
+        /// <param name="health">Detector Status</param>
+        /// <param name="message">Detector status message</param>
+        /// <example> 
+        /// This sample shows how to use <see cref="SetDetectorStatus"/> method.
+        /// <code>
+        /// public async static Task<![CDATA[<Response>]]> Run(DataProviders dp, OperationContext cxt, Response res)
+        /// {
+        ///     res.SetDetectorStatus(DetectorStatus.Critical, "This is an error message that will come from the detector");
+        /// }
+        /// </code>
+        /// </example>
+        public static void SetDetectorStatus(this Response response, DetectorStatus health, string message = null)
```

Comment:
nitpick : probably name the "health" parameter as "status" as the method says "SetDetectorStatus"

Also see my other comment of trying to deduce the status automatically **if the developer has not set any detector status explicitly**.

**Any Insight with Level Critical** : Detector Status is Critical, Message = "<Detector Name> found issues"
etc..